metricq_wizard_backend/metricq/cluster_scanner.py

```diff
@@ -0,0 +1,541 @@
+        try:
+            rate = metadata["rate"]
+            if not isinstance(rate, (int, float)):
+                logger.error(
+                    "Invalid rate: {} ({}) [{}]",
+                    rate,
+                    type(rate),
+                    metadata,
+                )
+            else:
+                allowed_age += Timedelta.from_s(1 / rate)
+        except KeyError:
+            # Fall back to compute tolerance from interval
+            with suppress(KeyError):
+                interval = metadata["interval"]
+                if isinstance(interval, str):
+                    allowed_age += Timedelta.from_string(interval)
+                elif isinstance(interval, (int, float)):
+                    allowed_age += Timedelta.from_s(interval)
+                else:
+                    logger.error(
+                        "Invalid interval: {} ({}) [{}]",
+                        interval,
+                        type(interval),
+                        metadata,
+                    )
+
+        try:
+            request_time = Timestamp.now()
+            result = await client.history_last_value(metric, timeout=60)
+
+            if result is None:
+                # No data points stored yet. This is bad.
+                await self.create_issue_report(
+                    scope_type="metric",
+                    scope=metric,
+                    issue_type="no_value",
+                    severity="warning",
+                    source=metadata.get("source"),
+                )
+                return
+
+            # we have a valid result, hence, there are stored data points.
+            await self.delete_issue_report(
+                scope_type="metric",
+                scope=metric,
+                issue_type="no_value",
+            )
+
+            age = request_time - result.timestamp
+
+            # Archived metrics are supposed to not receive new data points.
+            # For such metrics, the archived metadata is the ISO8601 string, when
+            # the metric was archived.
+            if age > allowed_age and not metadata.get("archived"):
+                # We haven't received a new data point in a while and the metric
+                # wasn't archived => It's dead, Jimmy.
+                await self.create_issue_report(
+                    scope_type="metric",
+                    scope=metric,
+                    issue_type="dead",
+                    severity="error",
+                    last_timestamp=str(result.timestamp.datetime.isoformat()),
+                    source=metadata.get("source"),
+                )
+                # if the metric is dead, it can't be undead as well
+                await self.delete_issue_report(
+                    scope_type="metric",
+                    scope=metric,
+                    issue_type="undead",
+                )
+            elif age <= allowed_age and metadata.get("archived"):
```

Comment:
Why don't we compare `result.timestamp` with `metadata["archived"]`?

Reply:
Would that be, in all honesty, helpful?

How would you deal with that? You would need to remove time points from the database, which is not easily doable. If you are simply moving the archived timestamp to a later date, you can just stick with the current check as well. However, moving the archived timestamp is neither intended nor implemented right now.

Reply:
I think we discussed this, but the solution is to change `metadata["archived"]`.